### DISCUSSION TITLE
Problem of using the field value of one module in another module

### DISCUSSION THREAD

**Discussion by EugeneCrown** [YetiForceCompany/YetiForceCRM#15680] [How to] [answered]
Dear colleagues!
My name is Eugene. I am from Krasnoyarsk, Siberia, Russia.

I have been using Yetiforce CRM for a long time for personal purposes and am very pleased with the product.
I ran into a problem when setting up a module that I created myself. Further explanations will be regarding this module.

The problem is this:
1) create a new order
![main](https://user-images.githubusercontent.com/98438973/151491610-cb116322-533d-48f5-9ae2-802739e057dc.JPG)
![choose good](https://user-images.githubusercontent.com/98438973/151491655-f5be2177-68c0-4f95-9896-c85227c90b3c.JPG)

2) I select a service ("Услуга печати"), attention to the service field "Price" ("Цена")
![focus on price of good](https://user-images.githubusercontent.com/98438973/151491739-21a50429-8a48-4a69-a016-fae4110cf5d8.JPG)

3) the service is selected, all is well. PROBLEM: I want the Service Price value to be placed in the "Print Price" field ("Услуга печати")
![need price from good](https://user-images.githubusercontent.com/98438973/151491766-30d67b78-24e7-4a54-9739-cc93b36382c9.JPG)

In the module, the "Print service" field ("Услуга печати") is configured as follows:
![price field](https://user-images.githubusercontent.com/98438973/151491788-93ca200b-15d2-4784-b627-7da1e17ff83c.JPG)

In the model, the "Print price" ("Цена печати") field is configured as follows:
![field for price from good](https://user-images.githubusercontent.com/98438973/151491807-146e7fc9-56d6-40ff-a716-baa52d5c9338.JPG)

ATTEMPTS TO SOLVING THE PROBLEM:
1) I try to implement through the workflow:
![workflow 1](https://user-images.githubusercontent.com/98438973/151491831-93385771-0b9c-4374-9026-07ca36b81522.JPG)

The workflow completes and substitutes the Service record ID from the database in the required field:
![workflow take ID of record from DB](https://user-images.githubusercontent.com/98438973/151491857-ca7fc073-c2e8-482b-8193-eebea18db578.JPG)

This is not what is needed.

2) I try to use a SQL query call to the database in the workflow expression:
![workflow with SQL-query](https://user-images.githubusercontent.com/98438973/151491878-59dd9b57-fcc0-4929-91e8-0d43e38494af.JPG)

This throws an error.

System parameters:
![system](https://user-images.githubusercontent.com/98438973/151491891-e009bfb9-6543-4a34-a575-6b50bade87a4.JPG)

Please help in solving the problem of using the field value of one module in another module.
Is it possible to use SQL queries in workflow expressions?
If there is no ready-made solution, can I order paid support to implement the functionality of using the field value of one module in another module?

**Comment by vovpff** [comment]
Вам нужно чтобы стоимость услуги заполнялась когда выбрали услугу или когда сохранили запись?

**Comment by EugeneCrown** [comment]
Мне стоимость нужна для дальнейших расчетов. Думаю, возможны оба варианта:
При выборе услуги - идеально.
Когда сохраняется запись - хотябы

**Comment by vovpff** [comment]
Первый случай - завтра попробую подсказать, когда до пк доберусь. Второй случай - я бы хендлер написал, который при сохранениее будет циферку обновлять.

Предлагаю завтра с первым случаем попробовать, а дальше как пойдет

**Comment by EugeneCrown** [comment]
Пробовать готов, очень жду ваших предложений.

**Comment by vovpff** [comment]
@EugeneCrown 
В этот файл https://github.com/YetiForceCompany/YetiForceCRM/blob/eff41081c5423cdf022e5137a2489b4fdd0857d5/app_data/moduleHierarchy.php#L74

Пишете, исходя из того, что основной модуль будет A, а услуга с подробностями будет B 

```
		'modulenameB' => [
			'filedname_usluga' => ['modulenameA' => ['fieldname_at_B']],
],
```

**Reply by EugeneCrown** [reply]
Tried to do it. I also tried the opposite variation (a->b).
This does not work.

               'IGRNC' => [
			'igrnid' => ['IGRN' => ['vendorid' => ['vendorid'], 'storageid' => ['storageid']]]
		],
		'IGDNC' => [
			'igdnid' => ['IGDN' => ['storageid' => ['storageid'], 'accountid' => ['accountid']]]
		],
		

		'IGDNC' => [
			'Orderbook' => ['service_price' => ['Printing' => ['printing_price']]],
		],

		
		'Assets' => [
			'contactid' => ['Contacts' => ['parent_id' => ['parent_id']]],
		],

Orderbook = Orders module (where you need to substitute the cost from the service)
Printing = A module that stores the services from which I want to get and substitute the price of the service
printing_price = Service module field storing the price
service_price = The field where I need to substitute the price from the service

**Comment by EugeneCrown** [comment]
Правильно понимаю: основной модуль будет A - это модуль, где я выбираю услугу и нужно отобразить значение цены ?

**Comment by vovpff** [comment]
@EugeneCrown дa
upd: скобка одна лишняя вроде.

**Comment by rskrzypczak** [comment]
Hello, 
I assume you're enjoying yourselves speaking in your native language, however, keep in mind that the default language in here is English. When you speak your own language you're limiting the number of participants of the conversation who could help you because they don't speak your language. 
I'm moving this to discussions as I guess there is no error in the app in here.

**Reply by vovpff** [reply]
@rskrzypczak sorry for Russian lang. I think if Eugeny solve his task he can write solution in English

**Comment by EugeneCrown** [comment]
I'm sorry. I will continue to write in English

**Comment by vovpff** [comment, marked as answer]
@EugeneCrown 
```
             1
	'IGDNC' => [
		     2                3                4                  5
              'Orderbook' => ['service_price' => ['Printing' => ['printing_price']]],
	],
```

1. your parent module name (module A)
2. field in module A what trigger update of price filed
3. related module B name 
4. Destination. field name in module A what should be filled with value from module B
5. Source field in module B. 

Your code little bit messed

In my system I Use this
```
		'HelpDesk' => [
            'product_id' => ['Products' => ['ethdrelvend' => ['vendor_id']]],
],
```

This mean what when I choose product in HelpDesk module field Vendor name filled with Vendor of product.

P.S. Welcome to tg group https://t.me/+QWA07Mgx6o5jNDRi

**Comment by EugeneCrown** [comment]
This solution helped. Thanks for the support!
